src/engine/strat_engine/device.rs

```diff
@@ -22,3 +34,29 @@ pub fn blkdev_size(file: &File) -> StratisResult<Bytes> {
         .map(|(_, res)| Bytes::from(res))
         .map_err(|e| e.into())
 }
+
+pub fn blkdev_logical_sector_size(file: &File) -> StratisResult<Bytes> {
+    let mut val = 0i32 as c_int; // util-linux uses int* as out-arg for ioctl
+    unsafe { blksszget(file.as_raw_fd(), &mut val) }.map_err(|e| {
+        StratisError::Msg(format!(
+            "Error reading logical sector size (BLKSSZGET): {}",
+            e
+        ))
+    })?;
+    // Allowed because the size should be less than u16::MAX
+    #[allow(clippy::cast_possible_truncation)]
+    Ok(Bytes::from(val as u16))
```

Comment:
Why are you using `as` when we have the macros for integer conversion and this method returns a `StratisResult` anyway?

Reply:
No good reason!